includes/class-wc-gateway-stripe.php

```diff
@@ -349,7 +398,25 @@ public function admin_scripts() {
 
 		$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
 
-		wp_register_script( 'woocommerce_stripe_admin', plugins_url( 'assets/js/stripe-admin' . $suffix . '.js', WC_STRIPE_MAIN_FILE ), [], WC_STRIPE_VERSION, true );
+		// Webpack generates an assets file containing a dependencies array for our built JS file.
+		$script_path       = 'build/upe_settings.js';
+		$script_asset_path = WC_STRIPE_PLUGIN_PATH . '/build/upe_settings.asset.php';
+		$script_url        = plugins_url( $script_path, WC_STRIPE_MAIN_FILE );
+		$script_asset      = file_exists( $script_asset_path )
+		? require( $script_asset_path )
+		: [ 'dependencies' => [] ];
```

Comment:
Maybe all of this could also go inside the `WC_Stripe_Features::is_upe_enabled()` conditional, so that it doesn't have to be executed when `WC_Stripe_Features::is_upe_enabled()` is falsy